lib/handler/redirect.c

```diff
@@ -24,17 +24,25 @@
 
 #define MODULE_NAME "lib/handler/redirect.c"
 
+typedef H2O_VECTOR(char *) char_vec;
+
 struct st_h2o_redirect_handler_t {
     h2o_handler_t super;
     int internal;
     int status;
-    h2o_iovec_t prefix;
+    h2o_iovec_vector_t prefix_list;
 };
 
 static void on_dispose(h2o_handler_t *_self)
 {
     h2o_redirect_handler_t *self = (void *)_self;
-    free(self->prefix.base);
+    if (self->prefix_list.entries != NULL) {
```

Comment:
I do not think you need this `if`. You can run the loop and `free` even when the list is empty (as both of them would be no-ops).

Reply:
done in https://github.com/h2o/h2o/pull/2186/commits/7405d448a51c179ab5668139ff539e3edddfe1f8